Author: Saleem Abdulrasool
Upstream: no
Reason: dont build all the extras, only gudev
--- 
--- udev-151/configure.ac	2010-03-03 11:10:06.457027031 +0000
+++ udev-151/configure.ac	2010-03-03 11:18:35.137186608 +0000
@@ -46,6 +46,16 @@
 	AC_DEFINE(ENABLE_LOGGING, [1], [System logging.])
 fi
 
+AC_ARG_ENABLE([gudev],
+              AS_HELP_STRING([--disable-gudev], [disable libgudev - libudev gobject interface]),
+	      [], [enable_gudev=yes])
+if test x"$enable_gudev" = x"yes" ; then
+    PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
+    AC_SUBST([GLIB_CFLAGS])
+    AC_SUBST([GLIB_LIBS])
+fi
+AM_CONDITIONAL([ENABLE_GUDEV], [test x"$enable_gudev" = x"yes"])
+
 AC_ARG_ENABLE([extras],
 	AS_HELP_STRING([--disable-extras], [disable extras with external dependencies]),
 	[], [enable_extras=yes])
@@ -55,10 +65,6 @@
 		AC_MSG_ERROR([gperf is needed])
 	fi
 
-	PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
-	AC_SUBST([GLIB_CFLAGS])
-	AC_SUBST([GLIB_LIBS])
-
 	AC_CHECK_LIB([acl], [acl_init], [:], AC_MSG_ERROR([libacl not found]))
 	AC_CHECK_HEADER([acl/libacl.h], [:], AC_MSG_ERROR([libacl header not found]))
 
--- udev-151/Makefile.am	2010-03-03 11:15:14.031152468 +0000
+++ udev-151/Makefile.am	2010-03-03 11:32:45.180028104 +0000
@@ -284,15 +284,9 @@
 libexec_PROGRAMS += extras/v4l_id/v4l_id
 dist_udevrules_DATA += extras/v4l_id/60-persistent-v4l.rules
 
-if ENABLE_EXTRAS
-# ------------------------------------------------------------------------------
-# conditional extras (need glib, libusb, libacl, ...)
-# ------------------------------------------------------------------------------
-dist_udevrules_DATA += \
-	rules/rules.d/75-net-description.rules \
-	rules/rules.d/75-tty-description.rules \
-	rules/rules.d/78-sound-card.rules
+BUILT_SOURCES =
 
+if ENABLE_GUDEV
 # ------------------------------------------------------------------------------
 # GUdev - libudev gobject interface
 # ------------------------------------------------------------------------------
@@ -330,7 +324,7 @@
 	extras/gudev/gudevmarshal.c \
 	extras/gudev/gudevenumtypes.h \
 	extras/gudev/gudevenumtypes.c
-BUILT_SOURCES = $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
+BUILT_SOURCES += $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
 
 extras_gudev_libgudev_1_0_la_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
@@ -407,6 +401,17 @@
 CLEANFILES += $(gir_DATA) $(typelibs_DATA)
 endif # ENABLE_INTROSPECTION
 
+endif # ENABLE_GUDEV
+
+if ENABLE_EXTRAS
+# ------------------------------------------------------------------------------
+# conditional extras (need libusb, libacl, ...)
+# ------------------------------------------------------------------------------
+dist_udevrules_DATA += \
+	rules/rules.d/75-net-description.rules \
+	rules/rules.d/75-tty-description.rules \
+	rules/rules.d/78-sound-card.rules
+
 # ------------------------------------------------------------------------------
 # Bluetooth HID devices with special magic to switch the device
 # ------------------------------------------------------------------------------
